x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/RemoteAccessAuthenticationService.java

```diff
@@ -70,12 +72,12 @@ public void authenticate(final String action, final TransportRequest request, fi
         final ThreadContext threadContext = authcContext.getThreadContext();
 
         // TODO revisit this once Node's Version is refactored
-        if (getMinNodeTransportVersion().before(Authentication.VERSION_REMOTE_ACCESS_REALM)) {
+        if (getMinNodeTransportVersion().before(VERSION_REMOTE_ACCESS_AUTHENTICATION)) {
```

Comment:
This was 8.7.0. I changed to avoid a 8.7 node receiving an authentication object of remote access type because:
1. There will be serialization and role building issues since 8.7 incorrectly assumes `List<RoleDescriptorBytes>` from the metadata
2. The whole feature won't be released till at least 8.8. So the criteria here does not really need to be 8.7. I also think the version to gate the whole feature is not necessary the same the version needed for low level changes.